app/assets/javascripts/code/services/google/_map.js

```diff
@@ -135,36 +135,12 @@ export const map = (
       });
     },
 
-    drawMarker: function(latLngObj, optionInput, existingMarker, level){
-      if(!latLngObj) {
-        return;
-      }
-      var latlng = new google.maps.LatLng(latLngObj.latitude, latLngObj.longitude);
-      var icon = "/assets/location_marker0.png";
+    drawMarker: function({ position, title, zIndex, icon }){
+      var newMarker = new google.maps.Marker({ position, title, zIndex, icon });
 
-      if (level) {
-        icon = "/assets/location_marker" + level + ".png";
-      }
+      newMarker.setMap(this.get());
+      this.markers.push(newMarker);
```

Comment:
I would like to add test for that but I'm not sure how to mock `new google.maps.Marker`?

Reply:
Ping me for a chat later on about this if you want ;)